### ISSUE TITLE
Terragrunt command validation breaks some workflows

### ISSUE THREAD

**Issue by JeanFred** [gruntwork-io/terragrunt#2689]
Since #2676 (answering #438), Terragrunt validates the Terraform commands and exits early if it is incorrect.

This breaks one, well, creative workflows I had that leverages hooks. Basically, I want to run semi-arbitrary commands in the terragrunt context, and I’m (ab)using before-hooks and non-existing commands.

My particular use-case is: in my team we have found convenient to be able to "lock" a Terragrunt environment − someone is doing some hairy work and no-one else should apply that layer in the meantime. We (ab)use the Terraform lock feature by placing in DynamoDB a lock object (with a nice message in there « locked by Alice because of issue #XYZ » ). But to do that I need to know the region, the DynamoDB table name etc − information that Terragrunt has (ie constructs from various variables etc, work which I certainly don't want to redo)

My solution was thus to use hooks (see below) on commands that do not exist. The before-hook does its job (locking or unlocking), and then Terragrunt fails because the command does not exist but that's fine.

That does not work anymore since #2676 : Terragrunt exits early.

Soooo, not sure what's the best way forward here. My use-case is a bit too hacky and specific to give it first-class support, I think ; How about an environment variable to be able to ignore the validation ?

[1] Full dump of hooks, in case anyone is interested:
```
  before_hook "lock_layer" {
    # Hook on fictional command to lock the layer
    commands = ["lock"]
    execute = [
      "python",
      "${get_repo_root()}/scripts/layer_locking.py",
      "create",
      "--region",
      local.aws_region,
      "--table",
      local.dynamodb_table,
      "--lock_id",
      "${local.bucket}/${local.key}"
    ]
  }

  before_hook "unlock_layer" {
    # Hook on fictional command to unlock the layer
    commands = ["force-unlock"]
    execute = [
      "python",
      "${get_repo_root()}/scripts/layer_locking.py",
      "delete",
      "--region",
      local.aws_region,
      "--table",
      local.dynamodb_table,
      "--lock_id",
      "${local.bucket}/${local.key}"
    ]
  }

  error_hook "check_layer_lock" {
    # The command failed as it could not acquire the state lock, let's see what's in the lock table
    commands = get_terraform_commands_that_need_locking()
    on_errors = [
      ".*Error locking state.*",
      ".*Error acquiring the state lock.*",
    ]
    execute = [
      "python",
      "${get_repo_root()}/scripts/layer_locking.py",
      "read",
      "--region",
      local.aws_region,
      "--table",
      local.dynamodb_table,
      "--lock_id",
      "${local.bucket}/${local.key}"
    ]
  }
```

**Comment by JeanFred** [comment]
(And yes, the similarity to https://xkcd.com/1172/ has not escaped me :) )

**Comment by levkohimins** [comment]
Resolved in [v0.50.11](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.50.11) release.
